crates/red_knot_python_semantic/resources/mdtest/annotations/never.md

```diff
@@ -0,0 +1,57 @@
+# NoReturn & Never
+
+## Annotation
+
+`NoReturn` to annotate functions that never return normally. `Never` represents the bottom type, a
+type that represents the empty set of Python objects. These two annotations can be used
+interchangeably.
+
+```py
+from typing import NoReturn, Never, Any
+
+# error: [invalid-type-parameter] "Type `typing.Never` expected no type parameter"
+x: Never[int]
+
+def stop() -> NoReturn:
+    raise RuntimeError("no way")
+
+# revealed: NoReturn
+reveal_type(stop())
+
+a1: NoReturn
+# TODO: Test `Never` is only available in python >= 3.11
+a2: Never
+b1: Any
+b2: int
+
+def f():
+    # revealed: NoReturn
+    reveal_type(a1)
+    # revealed: Never
+    reveal_type(a2)
+
+    # Never is compatible with all types.
```

Comment:
Let's use the specified language of "assignable to", not the unspecified and informal "compatible with".